cli/command/image/push.go

```diff
@@ -42,8 +46,16 @@ func NewPushCommand(dockerCli command.Cli) *cobra.Command {
 // RunPush performs a push against the engine based on the specified options
 func RunPush(dockerCli command.Cli, opts pushOptions) error {
 	ref, err := reference.ParseNormalizedNamed(opts.remote)
-	if err != nil {
+	switch {
+	case err != nil:
 		return err
+	case opts.all && !reference.IsNameOnly(ref):
+		return errors.New("tag can't be used with --all-tags/-a")
+	case !opts.all && reference.IsNameOnly(ref):
+		ref = reference.TagNameOnly(ref)
+		if tagged, ok := ref.(reference.Tagged); ok && !opts.quiet {
+			_, _ = fmt.Fprintf(dockerCli.Out(), "Using default tag: %s\n", tagged.Tag())
```

Comment:
nit: Was the linter complaining about the errors not checked for `fmt.Fprintf` ? 🤔

Reply:
I think we disabled the linter to check that, but my IDE still complained

We should create a utility for this to use everywhere (something like `printStdOut()` `printStdErr()`)